sklearn/svm/_classes.py

```diff
@@ -271,6 +296,22 @@ def fit(self, X, y, sample_weight=None):
         check_classification_targets(y)
         self.classes_ = np.unique(y)
 
+        if self.dual == "auto":
```

Comment:
Can we put this code in common with both SVR and SVC. We could call this function `_validate_dual_parameter`.